test/tap/tests/test_warnings-t.cpp

```diff
@@ -0,0 +1,347 @@
+/**
+ * @file test-warnings-t.cpp
+ * @brief This test will test warnings support in ProxySQL
+ */
+
+#include <stdio.h>
+#include <unistd.h>
+#include <string>
+#include <list>
+#include <tuple>
+#include <map>
+#include <thread>
+
+#include "json.hpp"
+#include <mysql.h>
+#include <mysql/mysqld_error.h>
+#include "tap.h"
+#include "command_line.h"
+#include "utils.h" 
+
+using LEVEL = std::string;
+using CODE = int;
+using MESSAGE = std::string;
+
+#define MYSQL_QUERY__(mysql, query) \
+	do { \
+		if (mysql_query(mysql, query)) { \
+			fprintf(stderr, "File %s, line %d, Error: %s\n", \
+					__FILE__, __LINE__, mysql_error(mysql)); \
+			goto cleanup; \
+		} \
+	} while(0)
+
+#define MYSQL_CLEAR_RESULT(mysql) do { MYSQL_RES* mysql_result = mysql_use_result(mysql); \
```

Comment:
Instead of `mysql_use_result()` , then looping through all the rows with `mysql_fetch_row()` , you can simply use `mysql_store_result()`:
https://dev.mysql.com/doc/c-api/8.0/en/mysql-store-result.html

At the end, use `mysql_free_result()` anyway.

Reply:
Done